src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    /// Utitlies class to manipulate MsBuild projects.
+    /// </summary>
+    internal abstract class MsBuildUtilities
+    {
+        /// <summary>
+        /// Gets a project property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="propertyName">Name of the property.</param>
+        /// <returns>Requested project property. Null if the property is not present.</returns>
+        public static ProjectPropertyElement GetProperty(ProjectRootElement project, string propertyName)
+        {
+            Requires.NotNull(project, "project");
+
+            return project.Properties
+                .Where(p => string.Equals(p.Name, propertyName, StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
```

Comment:
If we are updating the Property, shouldn't we be updating the last entry so that the evaluation gets the updated value?